bool ready
---
float32[] objs_x
float32[] objs_y
int32[]   ids
float32[] docks_x
float32[] docks_y